code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -201,6 +182,8 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 
 
 /obj/item/clothing/mask/cigarette/proc/handle_reagents()
+	var/amnt_smoked = smoke_speed * reagents.total_volume / reagents.maximum_volume
+	reagents.maximum_volume = max(reagents.maximum_volume - smoke_speed, 0)

```

Comment:
This means the cigarette will be finished in 72 life cycles, instead of 300 with the current system.